Bottom: 4cbcef0c64b91cf4786ce790f5290423575ec210
Top:    c5d8c083048d4d4a84a35fc0f538316070a55de7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:51:38 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index ddbe6ed..ed41e59 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -405,8 +405,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
